src/foam/java/refinements.js

```diff
@@ -932,6 +932,45 @@ foam.CLASS({
     ['javaJSONParser', 'new foam.lib.json.MapParser()'],
     ['javaInfoType', 'foam.core.AbstractObjectPropertyInfo'],
     ['javaFactory', 'return new java.util.HashMap();']
+  ],
+
+  methods: [
```

Comment:
Wouldn't it be less code to create AbstractMapPropertyInfo which extends AbstractObjectPropertyInfo and just has this method defined once so that we don't have multiple generated copies?